### PR TITLE
Add gcm-filters recipe

### PR DESCRIPTION
**Pull request by rabernat**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by rabernat** [comment]
@NoraLoose and @iangrooms, can you confirm that you're willing to be listed as maintainers here?

**Comment by iangrooms** [comment]
Not sure if I have the skills to be a maintainer.

**Comment by rabernat** [comment]
Usually all that is required is to click "merge" on an automatically generated pull request to approve a version upgrade. But if you want to be removed, just say so and I'll remove you. All things being equal, more maintainers is better.

**Comment by iangrooms** [comment]
That sounds fine. I was worried I'd be on the hook for more complicated stuff. You can include me as a maintainer.

File: recipes/gcm-filters/LICENSE.txt
Comment (dopplershift) [inline recipes/gcm-filters/LICENSE.txt:1]: Since this is already included in the PyPI tarball, better to rely on getting it from there rather than including it in the recipe. Can you remove from the PR?

**Review by dopplershift — CHANGES_REQUESTED** [review]
Recipe is in great shape, just remove the superfluous license file. Also need confirmation from @NoraLoose.

File: recipes/gcm-filters/LICENSE.txt
Comment (rabernat) [reply recipes/gcm-filters/LICENSE.txt:1]: Ok thanks for that. I guess I was misinterpreting the "License file is packaged" checkbox to mean we should include it here.

Comment (dopplershift) [reply recipes/gcm-filters/LICENSE.txt]: Yeah, that language is slightly ambiguous. It means "make sure it appears in the end result". Essentially, point `license_file` to the name of the file. If it's in the tarball, great! Otherwise, it will error and you need add it manually to the recipe (and for bonus points let upstream know).

**Review by dopplershift — APPROVED** [review]
Looks good. Just waiting on confirmation from @NoraLoose.

**Comment by NoraLoose** [comment]
Yes, I confirm that I am willing to be listed as a maintainer here.

**Pull request merged**
